pkg/cassandra/nodetool/nodetool.go

```diff
@@ -0,0 +1,170 @@
+package nodetool
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pborman/uuid"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/jetstack/navigator/pkg/cassandra/nodetool/client"
+)
+
+type NodeState string
+
+const (
+	NodeStateUnknown NodeState = "Unknown"
+	NodeStateNormal  NodeState = "Normal"
+	NodeStateLeaving NodeState = "Leaving"
+	NodeStateJoining NodeState = "Joining"
+	NodeStateMoving  NodeState = "Moving"
+)
+
+type NodeStatus string
+
+const (
+	NodeStatusUnknown NodeStatus = "Unknown"
+	NodeStatusUp      NodeStatus = "Up"
+	NodeStatusDown    NodeStatus = "Down"
+)
+
+type Node struct {
+	Host   string
+	ID     uuid.UUID
+	State  NodeState
+	Status NodeStatus
+	Local  bool
+}
+
+type NodeMap map[string]*Node
+
+func (nm NodeMap) LocalNode() *Node {
+	for _, node := range nm {
+		if node.Local {
+			return node
+		}
+	}
+	return nil
+}
+
+type Interface interface {
+	Status() (NodeMap, error)
```

Comment:
Could we rename `Status` to `Nodes`? Status is becoming quite an overloaded term in this package.

Reply:
Good idea, but again, I wanted to try and match nodetool as closely as possible.
This is intended to give a structured version of the `nodetool status` output.
Let's leave it as is for now and I can refactor later when I've got a better feeling for what operations we need to perform.

Reply:
👍